cvmfs/catalog.h

```diff
@@ -231,6 +231,7 @@ class Catalog : public SingleCopy {
   Counters& GetCounters() { return counters_; }
 
   inline const CatalogDatabase &database() const { return *database_; }
+  inline       CatalogDatabase &database()       { return *database_; }
```

Comment:
Those are *protected* accessors to be used by subclasses of `Catalog`. For example all `Sql` classes take a `Database&` in their constructors. I believe rather should the `database_` member in `Catalog` not be a pointer as it is now.

I need to look into the feasibility of this change, since currently one can have a `Catalog` object without an attached `Database`, which I find odd anyway. But I believe it was the reason for defining the `database_` member as a pointer.